src/reducers/app.js

```diff
@@ -16,52 +16,66 @@ const app = (state = {showSettings:'all', saySettings:'demand'}, action) => {
   let json, value;
   switch (action.type) {
     case UPDATE_PAGE:
-      return {
-        ...state,
-        page: action.page
-      };
+      state.page = action.page;
+      return Object.assign({}, state);
```

Comment:
`state.page = action.page`
This is mutating the state. What you should do is something like this:
```
return Object.assign({}, state, {page: action.page});
```

Reply:
oh i thought since i was returning a copy it was technically ok. ooooops fixing everywhere